### PR TITLE
refac: BatchCreatedDomainEvent -> Mediatr

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description

This PR will handle the domain event of a batch created with Mediatr.

- StartCalculationEndpoint has been removed

## References

* #972

### CODE REVIEW

File: source/dotnet/wholesale-api/Domain/Domain.csproj
Comment (energinet-ajw) [inline source/dotnet/wholesale-api/Domain/Domain.csproj:1]: As Bjarke mentioned in the previous PR. If the transaction fails after the Databricks job has started; there will be a job we don't know about. I think the solution is like Batmans, that is, an internal queue (Outbox for special domain events).

Comment (lasrinnil) [reply source/dotnet/wholesale-api/Domain/Domain.csproj:1]: This will be required in upcomming work. Lets keep events in the cache to begin with.

**Review by energinet-ajw — COMMENTED** [review]
Just a thought. Should DomainEventContainer be renamed?
"Container" is a new term (probably) from PetStore :)

**Comment by lasrinnil** [comment]
> 

Sure, lets not add to the list of "things" we use, I've renamed it to `DomainEventsRepository`, perhaps you have a better name?

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**